java/src/main/java/com/google/appengine/tools/mapreduce/MapReduceServlet.java

```diff
@@ -70,7 +71,7 @@ public class MapReduceServlet extends HttpServlet {
   public void init() throws ServletException {
     super.init();
     if (this.component == null) {
-      component = DaggerJobRunServiceComponent.create();
```

Comment:
so Dagger2 creates fresh copy each time, despite `@Singleton` annotation on the component.